app/models/mechanic.rb

```diff
@@ -0,0 +1,8 @@
+class Mechanic < ApplicationRecord
+  has_many :ride_mechanics
+  has_many :rides, through: :ride_mechanics
+
+  def add_ride(ride_id)
```

Comment:
This works with the `update` action in your `mechanics_controller` but should have rides through the associations vs using Ruby.